.github/workflows/run_coverage.yml

```diff
@@ -0,0 +1,51 @@
+name: Run Coverage
+on: [pull_request]
+jobs:
+  test:
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - name: Install poetry
+        run: |
+          pip install poetry
+      - name: Syncify Client
+        run: |
+          python scripts/syncronizer.py
+      - name: Start Server
+        run: |
+          cd api
+          poetry install --no-root
+          poetry run uvicorn src.main:app &
+          sleep 5
+          cd ..
+        env:
+          DATABASE_TYPE: sqlite
```

Comment:
With the addition of the pgvector stuff sqlite may not work. There is a docker-compose in `main` that has a postgresql instance with the pgvector extension in it. May need to do some kind of docker-compose type testing as well for testing